src/STEMMUS_SCOPE.m

```diff
@@ -815,7 +797,38 @@
             hSAVE = hN;
         end
     end
-    run Forcing_PARM;
+    %% Reconstruct the driving forces to be a continuous function of time
+    if TIMEOLD == KT
+        Ta(KT) = 0;
+        HR_a(KT) = 0;
+        Ts(KT) = 0;
+        U(KT) = 0;
+        SH(KT) = 0;
+        Rns(KT) = 0;
+        Rnl(KT) = 0;
+        Rn(KT) = 0;
+        TopPg(KT) = 0;
+        h_SUR(KT) = 0;
+    end
+    if NBCT == 1 && KT == 1
+        Ts(1) = 0;
+    end
+
+    ForcingData.WS_msr(ForcingData.WS_msr < 0.05) = 0.05;
+    Ta(KT) = ForcingData.Ta_msr(KT);
+    HR_a(KT) = 0.01 .* (ForcingData.RH_msr(KT));
+    U(KT) = 100 .* (ForcingData.WS_msr(KT));
+    Rns(KT) = (ForcingData.Rns_msr(KT)) * 8.64 / 24 / 100 * 1;
+    TopPg(KT) = 100 .* (ForcingData.Pg_msr(KT));
+    Ts(KT) = Tss;
+    Rn(KT) = (ForcingData.Rn_msr(KT)) * 8.64 / 24 / 100 * 1;
+    Precip(KT) = ForcingData.Precip_msr(KT) * 0.1 / DELT;
+    Gvc(KT) = ForcingData.LAI_msr(KT);
```

Comment:
Shouldn't these steps happen during initialization? And without the `(KT)` indexing?

Reply:
> 

Hi Bart, (KT) is needed. for each time step, forcing data at that particular time step (KT) should be used to drive the model.

Reply:
some lines are fixing e.g. `ForcingData.WS_msr(ForcingData.WS_msr < 0.05) = 0.05;`, some are unit conversion e.g. `TopPg(KT) = 100 .* (ForcingData.Pg_msr(KT));`, some are variable definition e.g. `Ts(KT) = Tss;`, and some are adaptive time step e.g.  `Rn(KT) = (ForcingData.Rn_msr(KT)) * 8.64 / 24 / 100 * 1;` that are not used in the model.

Reply:
> and some are adaptive time step e.g. `Rn(KT) = (ForcingData.Rn_msr(KT)) * 8.64 / 24 / 100 * 1;`

Hi Sarah, this is also unit conversion, not an adaptive time step.